### PR TITLE
Azure - Extend Firewall Rules Filter to CosmosDB and KeyVault

### PR DESCRIPTION
**Pull request by anastasiia-zolochevska**

### CODE REVIEW

**Comment by anastasiia-zolochevska** [comment]
#4106

File: docs/source/azure/policy/resources/cosmosdb.rst [hunk 1]
```diff
@@ -11,6 +11,10 @@ Filters
     - Metric Filter - Filter on metrics from Azure Monitor - (see `Cosmos DB Supported Metrics <https://docs.microsoft.com/en-us/azure/monitoring-and-diagnostics/monitoring-supported-metrics#microsoftdocumentdbdatabaseaccounts/>`_)
     - Tag Filter - Filter on tag presence and/or values
     - Marked-For-Op Filter - Filter on tag that indicates a scheduled operation for a resource
+- ``firewall-rules``
+    Firewall Rules Filter (see :ref:`azure_filters`)
+
+    .. c7n-schema:: azure.sqlserver.filters.firewall-rules
```
Comment (stefangordon) [inline docs/source/azure/policy/resources/cosmosdb.rst]: @logachev where did we end up on dashes? We have a few already.  Are we avoiding them or is this fine?

File: tools/c7n_azure/tests/azure_common.py [hunk 2]
```diff
@@ -179,6 +179,8 @@ def _response_callback(self, response):
                                response['headers'].items()
                                if k.lower() not in self.FILTERED_HEADERS}
 
+        response['headers']['content-location'] = [AzureVCRBaseTest._replace_subscription_id(cl) for cl in response['headers']['content-location']]
```
Comment (stefangordon) [inline tools/c7n_azure/tests/azure_common.py]: Does this header actually do anything for us?  Another option is to just add it to the filtered header list at the top of the file so it disappears.  This seems fine too though.

File: tools/c7n_azure/tests/cassettes/KeyVaultTest.test_firewall_rules_not_include_cidr.json [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+                    ]
+                },
+                "body": {
+                    "data": {
+                        "id": "/subscriptions/ea42f556-5106-4743-99b0-c129bfa71a47/resourceGroups/anzoloch-test-cc/providers/Microsoft.KeyVault/vaults/anzoloch-cc-kv1",
+                        "name": "anzoloch-cc-kv1",
+                        "type": "Microsoft.KeyVault/vaults",
+                        "location": "westus",
+                        "tags": {
+                            "CreatorEmail": "anzoloch@microsoft.com"
```
Comment (stefangordon) [inline tools/c7n_azure/tests/cassettes/KeyVaultTest.test_firewall_rules_not_include_cidr.json]: Having your email in here a bunch probably just gets you spam :) 

Also these cassettes are much larger than they should be - We have been bad about getting automation in place to keep the recording subscription clean.

I would just go delete all those key vaults so you only have the one provided by the ARM template, then record this again.

File: docs/source/azure/policy/resources/cosmosdb.rst (hunk 1 above)
Comment (stefangordon) [reply docs/source/azure/policy/resources/cosmosdb.rst]: err, nevermind - I see the original one was already named this, so if we are renaming it then its breaking change.

File: tools/c7n_azure/tests/azure_common.py (hunk 2 above)
Comment (anastasiia-zolochevska) [reply tools/c7n_azure/tests/azure_common.py]: Good point. Filtered it out.

File: tools/c7n_azure/tests/cassettes/KeyVaultTest.test_firewall_rules_not_include_cidr.json (hunk 3 above)
Comment (anastasiia-zolochevska) [reply tools/c7n_azure/tests/cassettes/KeyVaultTest.test_firewall_rules_not_include_cidr.json]: :) Thanks! Rerecorded cassettes

**Pull request merged**